src/datasets/arrow_dataset.py

```diff
@@ -3802,12 +3796,48 @@ def sort(
         if len(self) == 0:
             return self
 
-        # Check the column name
-        if not isinstance(column, str) or column not in self._data.column_names:
+        # Deprecation warning
+        if kind != "deprecated":
+            warnings.warn(
+                "The 'kind' parameter is deprecated and will be removed in future releases.",
+                category=DeprecationWarning,
+            )
+
+        # Check proper format of and for duplicates in column_names
+        if not isinstance(column_names, list):
+            column_names = [column_names]
+        if not len(column_names) == len(set(column_names)):
             raise ValueError(
-                f"Column '{column}' not found in the dataset. Please provide a column selected in: {self._data.column_names}"
+                "At least one of the column names in column_names occurs more than once. Ensure each column name is unique."
```

Comment:
I think we can remove this check and delegate it to `sort_indices`

Reply:
Could you elaborate on what you mean by "_delegate it to `sort_indices`_"?

Reply:
I don't think it's possible for `pc.sort_indices` to throw this error. Can you paste the full error stack trace?

Reply:
I didn't write that this error is thrown by `pc.sort_indices`. 
This was just to illustrate that calling sort in this way should be caught/prevented. The error occurs in `table`'s `_interpolation_search`.

Full error stack:
```
Traceback (most recent call last):
  File "c:\Users\miche\Documents\GitHub\test\testing.py", line 24, in <module>
    another_sorted_ds = ds.sort(["label", "label"], reverse=[True, False])
  File "C:\Users\miche\anaconda3\envs\hugging_face\lib\site-packages\datasets\arrow_dataset.py", line 514, in wrapper
    out: Union["Dataset", "DatasetDict"] = func(self, *args, **kwargs)
  File "C:\Users\miche\anaconda3\envs\hugging_face\lib\site-packages\datasets\fingerprint.py", line 480, in wrapper
    out = func(self, *args, **kwargs)
  File "C:\Users\miche\anaconda3\envs\hugging_face\lib\site-packages\datasets\arrow_dataset.py", line 3971, in sort
    return self.select(
  File "C:\Users\miche\anaconda3\envs\hugging_face\lib\site-packages\datasets\arrow_dataset.py", line 514, in wrapper
    out: Union["Dataset", "DatasetDict"] = func(self, *args, **kwargs)
  File "C:\Users\miche\anaconda3\envs\hugging_face\lib\site-packages\datasets\fingerprint.py", line 480, in wrapper
    out = func(self, *args, **kwargs)
  File "C:\Users\miche\anaconda3\envs\hugging_face\lib\site-packages\datasets\arrow_dataset.py", line 3563, in select
    return self._select_contiguous(start, length, new_fingerprint=new_fingerprint)
  File "C:\Users\miche\anaconda3\envs\hugging_face\lib\site-packages\datasets\arrow_dataset.py", line 514, in wrapper
    out: Union["Dataset", "DatasetDict"] = func(self, *args, **kwargs)
  File "C:\Users\miche\anaconda3\envs\hugging_face\lib\site-packages\datasets\fingerprint.py", line 480, in wrapper
    out = func(self, *args, **kwargs)
  File "C:\Users\miche\anaconda3\envs\hugging_face\lib\site-packages\datasets\arrow_dataset.py", line 3616, in _select_contiguous
    self.data.slice(start, length),
  File "C:\Users\miche\anaconda3\envs\hugging_face\lib\site-packages\datasets\table.py", line 1113, in slice
    return MemoryMappedTable(self.fast_slice(offset=offset, length=length), self.path, replays)
  File "C:\Users\miche\anaconda3\envs\hugging_face\lib\site-packages\datasets\table.py", line 139, in fast_slice
    i = _interpolation_search(self._offsets, offset)
  File "C:\Users\miche\anaconda3\envs\hugging_face\lib\site-packages\datasets\table.py", line 93, in _interpolation_search
    if arr[k] <= x < arr[k + 1]:
IndexError: only integers, slices (`:`), ellipsis (`...`), numpy.newaxis (`None`) and integer or boolean arrays are valid indices
```

Reply:
Check removed in https://github.com/huggingface/datasets/pull/5502/commits/53c0a537b5c5c55f5a2aa72715b39175cc690bf4

Reply:
Addressed in https://github.com/huggingface/datasets/pull/5502/commits/aabc95c4c4cfb3020d1356a018a3059a2914b0b1